source/reference/ejson/stringify.txt

```diff
@@ -0,0 +1,506 @@
+     -  Optional
+     -  Modifies output. If the value exists but it is not an array or a
+        function, ``EJSON.stringify`` returns all document fields. 
+ 
+        .. list-table::
+           :header-rows: 1
+           :widths: 30 70
+
+           * - Value
+             - Effect
+
+           * - array
+             - An array of document fields to include in the output
+
+           * - function
+             - A function that takes two parameters, ``key`` and
+               ``value``. ``key`` provides the function's ``this``
+               context. EJSON returns the transformed ``value``. See
+               below for an :ref:`example <ex-ejson-function>`   
+
+   * - ``spacer``
+     -  integer or string
+     -  Optional
+     -  Controls spacing in the output. Use ``null`` as a place holder
+        for ``replacer`` if you only want to specify the ``spacer``
+        option.
+ 
+        .. list-table::
+           :header-rows: 1
+           :widths: 30 70
+
+           * - Value
+             - Effect
+
+           * - integer
+             - The number of spaces to indent each level. ``10`` is the
+               maximum.
+
+           * - string
+             - A character to use to indent each level. This option
+               produces invalid JSON if you use a character other than a
+               space or tab. For more information see `JSON.stringify()
+               <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify>`__
+
+   * - ``options``
+     -  boolean
+     -  Optional
+     -  Additional configuration options
+      
+        .. list-table::
+           :header-rows: 1
+           :widths: 25 25 50
+
+           * - Option
+             - Default
+             - Meaning
+
+           * - ``relaxed``
+             - ``true``
+             - Enable relaxed mode for Extended JSON. Returns native 
+               JSON types instead of attaching BSON type information
+               where applicable.
+
+Behavior
+--------
+
+You can call the EJSON interface from inside an interactive ``mongosh``
+session or from the system command line using ``--eval``.
+
+Call the EJSON interface from an interactive session: 
+
+.. code-block:: javascript
+
+   EJSON.stringify( db.sales.find().toArray(), null, 2  )
+
+Call the EJSON interface from the system command line: 
+
+.. code-block:: shell
+
+   mongosh --eval "EJSON.stringify( db.sales.find().toArray(), null, 2  )"
+
+To control how documents are passed to EJSON, use one of the ``mongosh``
+:ref:`cursor method <doc-cursor-methods>` iterators. 
+
+.. list-table::
+   :header-rows: 1
+
+   * - Iterator
+     - Characteristics
+
+   * - :method:`.toArray() <cursor.toArray>`
+     - Blocking, buffers the entire result
+
+   * - :method:`.foreach <cursor.forEach()>`
+     - Non-blocking, prints documents one-by-one
+
+   * - :method:`.next() <cursor.next>`
+     -  Non-blocking, manually iterate over results
```

Comment:
[spacing nit]

Extra space between bullet and "Non-blocking".

Reply:
Done